databricks/koalas/frame.py

```diff
@@ -4371,7 +4371,11 @@ def drop(self, labels=None, axis=1,
                               for column, idx
                               in zip(self._internal.data_columns, self._internal.column_index)
                               if idx not in drop_column_index))
-            internal = self._internal.copy(data_columns=list(cols), column_index=list(idx))
+            internal = self._internal.copy(
+                sdf=self._sdf.select(
+                    self._internal.index_scols + [self._internal.scol_for(col) for col in cols]),
```

Comment:
We should use `idx` rather than `cols`.
Maybe we should rename `idx` as `idxes` or something, then:

```py
[self._internal.scol_for(idx) for idx in idxes]
```

Reply:
@ueshin it makes sense. i fixed them. Thanks!! :)